go/vt/vtgate/planbuilder/route_planning.go

```diff
@@ -390,30 +430,87 @@ func (rp *routePlan) planInOp(node *sqlparser.ComparisonExpr) (bool, error) {
 			return false, nil
 		}
 	}
-	opcode := func(*vindexes.ColumnVindex) engine.RouteOpcode { return engine.SelectIN }
+	opcode := func(v *vindexes.ColumnVindex) (vindexes.Vindex, engine.RouteOpcode) {
+		return v.Vindex, engine.SelectIN
+	}
 	return rp.haveMatchingVindex(node, column, value, opcode), err
 }
 
+func (rp *routePlan) planLikeOp(node *sqlparser.ComparisonExpr) (bool, error) {
+	column, ok := node.Left.(*sqlparser.ColName)
+	if !ok {
+		return false, nil
+	}
+
+	val, err := makePlanValue(node.Right)
+	if err != nil || val == nil {
+		return false, err
+	}
+
+	opcode := func(vindex *vindexes.ColumnVindex) (vindexes.Vindex, engine.RouteOpcode) {
+		if prefixable, ok := vindex.Vindex.(vindexes.Prefixable); ok {
+			return prefixable.PrefixVindex(), engine.SelectEqual
+		}
+
+		// if we can't use the vindex as a prefix-vindex, we can't use this vindex at all
+		return nil, engine.SelectScatter
+	}
+
+	return rp.haveMatchingVindex(node, column, *val, opcode), err
+}
+
+func (rp *routePlan) planIsExpr(node *sqlparser.IsExpr) (bool, error) {
+	// we only handle IS NULL correct. IsExpr can contain other expressions as well
+	if node.Right != sqlparser.IsNullOp {
+		return false, nil
+	}
+	column, ok := node.Left.(*sqlparser.ColName)
+	if !ok {
+		return false, nil
+	}
+	val, err := makePlanValue(&sqlparser.NullVal{})
+	if err != nil || val == nil {
+		return false, err
+	}
+
+	return rp.haveMatchingVindex(node, column, *val, equalOrEqualUnique), err
+}
+
+func makePlanValue(n sqlparser.Expr) (*sqltypes.PlanValue, error) {
+	value, err := sqlparser.NewPlanValue(n)
+	if err != nil {
+		// if we are unable to create a PlanValue, we can't use a vindex, but we don't have to fail
+		if strings.Contains(err.Error(), "expression is too complex") {
+			return nil, nil
+		}
+		// something else went wrong, return the error
+		return nil, err
+	}
+	return &value, nil
+}
+
 func (rp *routePlan) haveMatchingVindex(
-	node *sqlparser.ComparisonExpr,
+	node sqlparser.Expr,
 	column *sqlparser.ColName,
 	value sqltypes.PlanValue,
-	opcode func(*vindexes.ColumnVindex) engine.RouteOpcode,
+	opcode func(*vindexes.ColumnVindex) (vindexes.Vindex, engine.RouteOpcode),
```

Comment:
Small nit: To maintain `opcode`'s responsibility focused on getting the `RouteOpcode` and not the `Vindex` type + the `RouteOpcode`, we could add a second function as a parameter that will only return the `Vindex`